src/ExpManager.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import { IExperimentationService } from "vscode-tas-client";
+
+export class ExpManager {
+    private flightsLoaded: boolean = false;
+    private exp: IExperimentationService | undefined;
+
+    public initialize(exp: IExperimentationService | undefined) {
+        this.exp = exp;
+    }
+
+    public async isFlightEnabled(flight: string): Promise<boolean> {
+        if (!this.exp) {
+            return false;
+        }
+
+        try {
```

Comment:
The `vscode-tas-client` package will query the treatment assignment service in the background (every 30 minutes), so flight information should never be much staler than the last opened instance of VS Code. If you want to be absolutely sure your extension has up-to-date flight information, you can do this. However, it probably isn't necessary.

Reply:
So does that mean that we can just call `isCachedFlightEnabled()` here, and call `isFlightEnabledAsync()` for 1 time in the `init` method?

Reply:
You shouldn't ever need to call `isFlightEnabledAsync` unless you want the absolute latest flight information. But I can't think of any scenarios where you would.

Reply:
I see. Thank you @daytonellwanger.

Another question, is below code needed in the init functions?

```ts
const asyncDummyCheck = (arg: any) => {
        expService?.isFlightEnabledAsync('dummy').then((v) => { return; }).catch((r) => { return; });
    };
    expService?.isCachedFlightEnabled('dummy').then(asyncDummyCheck).catch(asyncDummyCheck);
}
```

Reply:
This should no longer be needed with the version of the `vscode-tas-client` package I mentioned above.

Reply:
I see. Thank you for your help @daytonellwanger. :)

I'll update the code according to the comments.